### PR TITLE
Add presubmits and periodics for Cluster API release-0.4

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

/assign @CecileRobertMichon @fabriziopandini @sbueringer

### CODE REVIEW

**Comment by vincepri** [comment]
/assign @dims 
for image pushing

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-main.yaml [hunk 1]
```diff
@@ -119,7 +114,6 @@ presubmits:
     branches:
     # The script this job runs is not in all branches.
     - ^master$
-    - ^operator-0.4$
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-main.yaml:122]: should the operator tests be part of main instead of release-0.4? 🤔

Comment (vincepri) [reply config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-main.yaml:122]: I left it there for now given that's targeting 0.4 branch

Comment (CecileRobertMichon) [reply config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-main.yaml:122]: right but it doesn't make sense to have that work be backported to 0.4 given that it's not even merged in the release-0.4 branch yet...

**Comment by sbueringer** [comment]
@vincepri I assume this means we won't run the upgrade tests for release-0.4? (they give us most of our Kubernetes version test coverage)

**Pull request merged**